history.go

```diff
@@ -7,37 +7,60 @@ import (
 )
 
 var (
-	ErrBadEpoch    = errors.New("[merkletree] Next epoch's STR has bad epoch")
-	ErrSTRNotFound = errors.New("[merkletree] STR not found")
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
 )
 
+type Epoch int64
+
 type History struct {
-	str           *SignedTreeRoot
-	epochInterval int64
+	currentSTR    *SignedTreeRoot
+	hashChain     map[Epoch]*SignedTreeRoot
+	epochs        []Epoch // slice of epochs in hashChain
+	epochInterval Epoch
```

Comment:
Since policies are epoch-specific, I think `UpdateHistory()` should have `Policies` as a parameter. I updated the code in a2ff2042fd80b36947392cccb6a49f33e328b458